### ISSUE TITLE
[Feature] Search using regular expressions

### ISSUE THREAD

**Issue by Lucifer25x** [zyrouge/symphony#622]
### Description

It'd be very good if we had ability to search for things using regular expressions.

### Solution

Regex could be used for searching exact titles or more detailed searches. (Which would make many things easier in some cases)

### Additional Context

https://github.com/zyrouge/symphony/issues/621
This feature could be a solution to this thing, too.

### Contribution Guidelines

- [X] I agree to follow the [Contribution Guidelines](https://github.com/zyrouge/symphony/wiki/Contributions-Guidelines#issues).

**Comment by controlrepo** [comment]
> We should be safe since our use case is no where being critical.

Of course, the worst that can happen is a DoS or an attempt of privilege escalation initiated by incorrectly parsed filename for example, no Internet blackout. I know, I'm not this daft.

What I'm saying is that regular expression handling is tricky and even the best, most experienced companies sometimes fail. There's whole class of exploits aimed at that and I'd recommend against even implementing this: https://ieeexplore.ieee.org/document/9519406

Example caught in the wild: https://ethicalhacking.uk/cve-2024-23346-arbitrary-code-execution-in-pymatgen-via-insecure/
